crates/nu-cli/src/utils/data_processing.rs

```diff
@@ -214,6 +214,80 @@ pub fn sum(data: Vec<Value>) -> Result<Value, ShellError> {
     Ok(acc)
 }
 
+pub fn max(data: Vec<Value>) -> Result<Value, ShellError> {
+    if data.is_empty() {
+        return Err(ShellError::unexpected(
+            "Attempted to find maximum in empty data",
+        ));
+    }
+
+    let mut biggest = if let UntaggedValue::Primitive(p) =
+        &data.first().expect("Expected a initial value").value
+    {
+        p.clone()
+    } else {
+        Value::zero()
+            .as_primitive()
+            .expect("Expected Value::zero to be primitive")
+    };
+
+    for value in data.into_iter() {
+        match value.value {
+            UntaggedValue::Primitive(p) => {
+                biggest = p.max(biggest);
```

Comment:
This code right here, could be the cause of the bug. Need to do a quick test to see if I can re-produce.